networkx/algorithms/link_prediction.py

```diff
@@ -275,11 +275,20 @@ def common_neighbor_centrality(G, ebunch=None, alpha=0.8):
            Sci Rep 10, 364 (2020).
            https://doi.org/10.1038/s41598-019-57304-y
     """
-    shortest_path = nx.shortest_path(G)
+
+    if not 0 <= alpha <= 1:
+        raise nx.NetworkXAlgorithmError("Alpha must be between 0 and 1")
```

Comment:
Do we need to restrict the values of alpha?
I know that traditionally the values would be between 0 and 1. But I've often seen algorithms with a weighted average like this be "over-compensated" by choosing weights that are either negative or more than one. This can sometimes successfully measure a feature.  It's like putting a negative weight on common_neighbors with compensating positive weight on closeness.  

If we're not sure, I'd prefer to add a doc_string line: "Values for alpha should normally be between 0 and 1."
and take out the explicit check and raise exception code.

Reply:
Good point, I had not considered that. I've removed the restriction code and the test for it. I've also added the sentence suggesting normal values to the doc string.